Added Favicon Support for React-Page

ammit opened the pull request:
I used the react-page logo for the favicon, and used connect middleware to include it.

**Favicon -** 
![favicon](https://f.cloud.github.com/assets/2105944/1049018/d88a8b7e-1098-11e3-8e80-b4864e8cac97.gif)

ammit committed:
added favicon support

ammit committed:
changed favicon to react-page

jordwalke commented:
Nice, @ammit! Did you get a chnace to sign the CLA mentioned here (https://github.com/facebook/react/blob/master/CONTRIBUTING.md)?

ammit commented:
@jordwalke Yes, I did it.

kassens commented on server.js:
```diff
@@ -74,6 +74,7 @@ if (!isServer) {
   var app = connect()
     .use(reactMiddleware.provide(buildOptions))
     .use(connect['static'](FILE_SERVE_ROOT))
+    .use(connect.favicon(path.join(FILE_SERVE_ROOT, 'elements/favicon/favicon.ico')))
```
`/` should not be hardcoded, use `path.join` for everything.

```
path.join(FILE_SERVE_ROOT, 'elements', 'favicon', 'favicon.ico')
```

ammit committed:
Removed hardCoded Path

Using path.join() instead of hard coded path for global support. 

> Hard coded path with slash (/) will break in windows which uses backslashes.

jordwalke commented:
@ammit  Merging, now this is looking like a _real_ page!
